contracts/PangolinExchange.sol

```diff
@@ -39,12 +47,14 @@ contract PangolinExchange is Ownable {
    * @dev _amount amount of the ERC20 token to be bought
    * TODO: Implement slippage % tolerance and add as a require check
   **/
-  function buyERC20Token(address _token, uint256 _amount) payable external RefundRemainder {
+  function buyAsset(bytes32 _token, uint256 _amount) payable external override RefundRemainder {
+    require(assetToAddressMapping[_token] != address(0), "Trading this asset is not allowed");
```

Comment:
We could extract this logic into a separate function - getAssetAddress() that will throw an error if asset name is not mapped (The same logic is used in 3 functions). Let's also test this.